### PR TITLE
fix(show): preserve None in output

### PR DESCRIPTION
**Pull request by shcheklein**
Instead of this output for empty File objects (e.g. after full outer merge):

```
    id   file  file file    file  file      file                      file     file right_id  \
       source  path size version  etag is_latest             last_modified location            
0  NaN   None  None  NaN    None  None       NaN                       NaT     None      5.0   
1  NaN   None  None  NaN    None  None       NaN                       NaT     None      6.0   
2  NaN   None  None  NaN    None  None       NaN                       NaT     None      7.0   
3  1.0            1  0.0                     1.0 1970-01-01 00:00:00+00:00     None      NaN   
4  2.0            2  0.0                     1.0 1970-01-01 00:00:00+00:00     None      NaN   
5  3.0            3  0.0                     1.0 1970-01-01 00:00:00+00:00     None      NaN 
```

(mind NaT and NaN)

we are getting real values from the database after this change, and we'll see `None`.

## Summary by Sourcery

Preserve Python None values in DataFrame outputs by introducing an object-backed `to_pandas(as_object=True)` mode and updating `show()` to leverage it, ensuring CLI output displays None rather than NaN or NaT.

New Features:
- Add an `as_object` option to `to_pandas` to emit object-typed DataFrames with Python None values

Bug Fixes:
- Make `show()` use the object-backed DataFrame so that None values are displayed instead of NaN/NaT

Tests:
- Add unit tests for `to_pandas(as_object=True)` to verify None preservation
- Add a functional test for `show()` to assert None is shown instead of NaN/NaT

### CODE REVIEW

**Review by 0x2b3bfa0 — APPROVED** [review]
NaR (not a review)

File: src/datachain/lib/dc/datachain.py [hunk 1]
```diff
@@ -1984,6 +1987,9 @@ def to_pandas(
             columns = pd.MultiIndex.from_tuples(map(tuple, headers))
 
         results = self.results(include_hidden=include_hidden)
+        if as_object:
+            df = pd.DataFrame(results, columns=columns, dtype=object)
+            return df.where(pd.notna(df), None)
```
[the remark this answers is not in the document]
Comment (shcheklein) [reply src/datachain/lib/dc/datachain.py:1992]: Addressed here https://github.com/iterative/datachain/pull/1372
